### PR TITLE
Add new Person(User) model for regular users

### PR DESCRIPTION
**Pull request by suricactus**
Since quite some time we have been planning to create a new model for individual users that inherits the base User model.

The commit messages describe what are the changes. 

TODOs:
- Add admin page for the new Person model

Note changing the `INVITATIONS_ACCEPT_INVITE_AFTER_SIGNUP` setting is a collateral damage since I changed the `ACCOUNT_ADAPTER` to custom adapter. Unfortunately this setting is available only if `ACCOUNT_ADAPTER = "invitations.models.InvitationsAdapter"`.

### CODE REVIEW

**Review by olivierdalang — COMMENTED** [review]
Looks fine so far, I thought it would require much much more changes in the code !

How much more change would it be to have only `Person` inherit AbstractUser ? This way we remove the confusing fact that teams and organizations have fields like passwords, last_login, etc.

**Comment by olivierdalang** [comment]
(about that last part of AbstractUser, I think it relates to your tentative of changing settings.AUTH_USER_MODEL)

File: docker-app/qfieldcloud/core/models.py [hunk 1]
```diff
@@ -295,6 +284,17 @@ def username_with_full_name(self) -> str:
     def has_geodb(self) -> bool:
         return hasattr(self, "geodb")
 
+    @property
+    def polymorph(self) -> "User":
```
[the remark this answers is not in the document]
Comment (suricactus) [reply docker-app/qfieldcloud/core/models.py:288]: Makes much more sense, thanks!

**Comment by suricactus** [comment]
> How much more change would it be to have only Person inherit AbstractUser ? This way we remove the confusing fact that teams and organizations have fields like passwords, last_login, etc.

I am afraid to estimate, but I would try it in a follow-up PR.

**Pull request merged**